bundles/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/handler/RokuHandler.java

```diff
@@ -0,0 +1,241 @@
+                        // When nothing playing, 'close' is reported, replace with 'stop'
+                        updateState(PLAY_MODE, new StringType(playerInfo.getState().replaceAll(CLOSE, STOP)));
+
+                        // Remove non-numeric from string, ie: ' ms'
+                        String position = playerInfo.getPosition().replaceAll(NON_DIGIT_PATTERN, EMPTY);
+                        if (!EMPTY.equals(position)) {
+                            updateState(TIME_ELAPSED,
+                                    new QuantityType<>(Integer.parseInt(position) / 1000, API_SECONDS_UNIT));
+                        } else {
+                            updateState(TIME_ELAPSED, UnDefType.UNDEF);
+                        }
+
+                        String duration = playerInfo.getDuration().replaceAll(NON_DIGIT_PATTERN, EMPTY);
+                        if (!EMPTY.equals(duration)) {
+                            updateState(TIME_TOTAL,
+                                    new QuantityType<>(Integer.parseInt(duration) / 1000, API_SECONDS_UNIT));
+                        } else {
+                            updateState(TIME_TOTAL, UnDefType.UNDEF);
+                        }
+                    } catch (RokuHttpException | NumberFormatException e) {
+                        logger.debug("Unable to retrieve Roku media-player info. Exception: {}", e.getMessage(), e);
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                    }
+                }
+            };
+            this.refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, refreshInterval, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * Start the job to periodically update list of apps installed on the the Roku
+     */
+    private void startAppListRefresh() {
+        ScheduledFuture<?> appListJob = this.appListJob;
+        if (appListJob == null || appListJob.isCancelled()) {
+            Runnable runnable = () -> {
+                synchronized (sequenceLock) {
+                    try {
+                        List<App> appList = communicator.getAppList();
+
+                        List<StateOption> appListOptions = new ArrayList<>();
+                        // Roku Home will be selected in the drop-down any time an app is not running.
+                        appListOptions.add(new StateOption(ROKU_HOME_ID, ROKU_HOME));
+
+                        appList.forEach(app -> {
+                            appListOptions.add(new StateOption(app.getId(), app.getValue()));
+                        });
+
+                        stateDescriptionProvider.setStateOptions(new ChannelUID(getThing().getUID(), ACTIVE_APP),
+                                appListOptions);
+
+                    } catch (RokuHttpException e) {
+                        logger.debug("Unable to retrieve Roku installed app-list. Exception: {}", e.getMessage(), e);
+                    }
+                }
+            };
```

Comment:
You can also put this into a method and than pass that method the scheduler with `this::<method name>`